### PR TITLE
workflows: add yaml-lint

### PR DESCRIPTION
**Pull request by Noki**
This adds yaml-lint to the workflow and ignores ansible vault files in the worker directories.

### CODE REVIEW

**Comment by Akira25** [comment]
@Noki How is the state of this? Is it still a draft?

**Comment by Noki** [comment]
@Akira25 ready now. Just wanted the fixes to be merged first.

File: .config/yaml-lint.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+# This extends the default yaml-lint conf by adjusting some options.
+
+extends: default
+
+# ignore possible venv directory and encrypted ansible vault files in the worker directories
+ignore: |
+  venv
+  inventory/host_vars/buildbot-worker-*/main.yml
```
Comment (Akira25) [inline .config/yaml-lint.yml:9]: From hostvars, we should add the buildbot-master too

from groupvars, we should add:
- all
- users
- buildbot

too.

**Review by Akira25 — CHANGES_REQUESTED** [review]
Looks already pretty good to me. We should just add some more encrypted files.

File: .config/yaml-lint.yml (hunk 1 above)
Comment (Noki) [reply .config/yaml-lint.yml:9]: they don't end with .yml or yaml and don't need to be included.

Comment (Akira25) [reply .config/yaml-lint.yml:9]: right. Then this looks good to me.

**Review by Akira25 — APPROVED** [review]

**Pull request merged**